Upgrade python

mbertrand opened the pull request:
#### Pre-Flight checklist

- [ ] Screenshots and design review for any changes that affect layout or styling
  - [ ] Desktop screenshots
  - [ ] Mobile width screenshots
- [ ] Migrations
  - [ ] Migration is backwards-compatible with current production code
- [ ] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested
- [ ] Settings
  - [ ] New settings are documented and present in `app.json`
  - [ ] New settings have reasonable development defaults, if applicable

#### What are the relevant tickets?
Closes https://github.com/mitodl/open-discussions/issues/3724

#### What's this PR do?
(Required)

#### How should this be manually tested?
(Required)

#### Where should the reviewer start?
(Optional)

#### Any background context you want to provide?
(Optional)

#### Screenshots (if appropriate)
(Optional)

#### What GIF best describes this PR or how it makes you feel?
(Optional)

abeglova reviewed (approved):
LGTM Everything i tested worked great

mbertrand commented:
Additional notes:

- Updated celery/sentry to get rid of an excepton logging exception (https://github.com/getsentry/sentry-python/issues/1798)

- Updated attrs to get rid of a warning when running backpopulate_<course>_data commands

- Updated google-api-python-client to fix youtube video syncing

- Updated pylint to avoid false import errors with os, io
